src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java

```diff
@@ -16,153 +16,188 @@
+									+ notificationMessage);
+				} else if (testKey != null && !testKey.isEmpty()) {
+					throw new SubscriptionValidationException(
+							"Key in notificationMessage must be empty when NOT using key/value pairs: "
+									+ notificationMessage);
+				} else if (StringUtils.isBlank(testValue)) {
+					throw new SubscriptionValidationException(
+							"Value in notificationMessage must have a value when NOT using key/value pairs: "
+									+ notificationMessage);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Validation of notificationMeta parameter Throws
+	 * SubscriptionValidationException if validation of the parameter fails due to
+	 * wrong format of parameter.
+	 *
+	 * @param notificationMeta
+	 */
+	private void validateNotificationMeta(String notificationMeta) throws SubscriptionValidationException {
+		String regex = ".*[\\s].*";
+		if (notificationMeta == null) {
+			throw new SubscriptionValidationException("Required field NotificationMeta has not been set");
+		} else if (Pattern.matches(regex, notificationMeta)) {
+			throw new SubscriptionValidationException("Wrong format of NotificationMeta: " + notificationMeta);
+		}
+	}
+
+	/**
+	 * Validation of notificationType parameter Throws
+	 * SubscriptionValidationException if validation of the parameter fails due to
+	 * wrong format of parameter.
+	 *
+	 * @param notificationType
+	 */
+	private void validateNotificationType(String notificationType) throws SubscriptionValidationException {
+		String regexMail = "[\\s]*MAIL[\\\\s]*";
+		String regexRestPost = "[\\s]*REST_POST[\\\\s]*";
+		if (notificationType == null) {
+			throw new SubscriptionValidationException("Required field NotificationType has not been set");
+		} else if (!(Pattern.matches(regexMail, notificationType)
+				|| Pattern.matches(regexRestPost, notificationType))) {
+			throw new SubscriptionValidationException("Wrong format of NotificationType: " + notificationType);
+		}
+	}
+
+	private void RestPostMediaType(String restPostMediaType) throws SubscriptionValidationException {
+		String regexApplication_JSON = "[\\s]*application/json[\\\\s]*";
+		String regexApplicationFormUrlEncoded = "[\\s]*application/x-www-form-urlencoded[\\\\s]*";
+		if (restPostMediaType == null) {
+			throw new SubscriptionValidationException("Required field RestPostMediaType has not been set");
+		} else if (!(Pattern.matches(regexApplication_JSON, restPostMediaType)
+				|| Pattern.matches(regexApplicationFormUrlEncoded, restPostMediaType))) {
+			throw new SubscriptionValidationException("Wrong format of RestPostMediaType: " + restPostMediaType);
+		}
+	}
+
+	/**
+	 * Validation of email address Throws SubscriptionValidationException if
+	 * validation of the parameter fails due to wrong format of parameter.
+	 *
+	 * @param email
+	 */
+	public void validateEmail(String email) throws SubscriptionValidationException {
+		final Pattern VALID_EMAIL_ADDRESS_REGEX = Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$",
```

Comment:
In the different validation functions I've seen different namings of these variables and personally I like this naming convention best, please change the others to follow this pattern.

Reply:
It is not part of my commit

Reply:
You can click 'Diff settings' at the top of the 'Files changed' tab and tick the box 'Hide whitespace changes'. This hides any indentation changes made. You will still see break lines caused by reformatting though.

Reply:
Still, I agree that a description of the change in the PR template is required.